store/mongo/migration_2_0_0.go

```diff
@@ -104,50 +93,59 @@ func (m *migration_2_0_0) Up(from migrate.Version) error {
 	ctx = identity.WithContext(ctx, &identity.Identity{
 		Tenant: tenantID,
 	})
+	writes := make([]mongo.WriteModel, 0, findBatchSize)
 
-	for collection := range collections {
-		// get all the documents in the collection
+	for collection, idxes := range collections {
+		writes = writes[:0]
 		findOptions := mopts.Find().
 			SetBatchSize(findBatchSize).
 			SetSort(bson.D{{Key: dbFieldID, Value: 1}})
 		coll := client.Database(m.db).Collection(collection)
 		collOut := client.Database(DbName).Collection(collection)
-		_, err := collOut.Indexes().CreateMany(ctx, collections[collection].Indexes)
-		if err != nil {
-			return err
+		if len(idxes.Indexes) > 0 {
+			_, err := collOut.Indexes().CreateMany(ctx, collections[collection].Indexes)
```

Comment:
if m.db == DbName then coll is collOut, so do we need to create them in any case? shall we remove the indices and then create them?

Reply:
Oops.. This was only supposed to execute when `m.db == DbName`. We don't have to drop existing indexes if the keys and index names don't conflict with the ones created here.
Good catch, I will move this block inside the next if clause.